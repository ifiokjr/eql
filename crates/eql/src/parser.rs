use edgeql_parser::tokenizer::SpannedToken;
use edgeql_parser::tokenizer::TokenStream;

pub fn tokenize(input: &str) -> Vec<SpannedToken> {
  let mut token_stream = TokenStream::new(input);
  let mut tokens = Vec::new();

  for token_result in &mut token_stream {
    match token_result {
      Ok(token) => tokens.push(token),
      Err(_error) => {
        println!("Error: {:?}", _error);
        // handle errors here
      }
    }
  }

  tokens
}

#[cfg(test)]
mod tests {
  use crate::parser::tokenize;

  #[test]
  fn simple_module_tokens() {
    assert_eq!(4, 4);
    let tokens = tokenize("module default {}");
    insta::assert_debug_snapshot!(tokens);
  }
}
