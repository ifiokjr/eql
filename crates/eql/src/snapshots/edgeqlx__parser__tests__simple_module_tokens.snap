---
source: crates/eql/src/parser.rs
expression: tokens
---
[
    SpannedToken {
        token: Token {
            kind: Keyword,
            value: "module",
        },
        start: Pos(1:1),
        end: Pos(1:7),
    },
    SpannedToken {
        token: Token {
            kind: Ident,
            value: "default",
        },
        start: Pos(1:8),
        end: Pos(1:15),
    },
    SpannedToken {
        token: Token {
            kind: OpenBrace,
            value: "{",
        },
        start: Pos(1:16),
        end: Pos(1:17),
    },
    SpannedToken {
        token: Token {
            kind: CloseBrace,
            value: "}",
        },
        start: Pos(1:17),
        end: Pos(1:18),
    },
]
